test/specs/components/Slot/Slot-test.ts

```diff
@@ -1,6 +1,69 @@
-import { isConformant } from 'test/specs/commonTests'
+import { mount } from 'enzyme'
+
+import * as lib from 'src/lib'
 import Slot from 'src/components/Slot'
+import { createSlotFactory } from 'src/components/Slot/Slot'
+import { isConformant } from 'test/specs/commonTests'
 
 describe('Slot', () => {
-  isConformant(Slot, { exportedAtTopLevel: false })
+  const createSlot = (factoryFn: Function, val, options?) =>
+    mount(factoryFn(val, options)).find(Slot)
+
+  describe('is conformant', () => {
+    isConformant(Slot, { exportedAtTopLevel: false })
+  })
+
+  describe('createSlotFactory', () => {
+    it('calls createShorthand from lib', () => {
+      const mapValueToProp = mappedProp => ({ mappedProp })
+      const value = 'testValue'
+      const createShorthandSpy = spyOn(lib, 'createShorthand')
+      createSlotFactory('span', mapValueToProp)(value)
+
+      expect(createShorthandSpy).toHaveBeenCalledWith(
+        Slot,
+        mapValueToProp,
+        value,
+        expect.any(Object),
+      )
+    })
+
+    it('sets correct "as" prop in defaultProps', () => {
+      const as = 'span'
+      const options = { testOption: 'test option value' }
+      const createShorthandSpy = spyOn(lib, 'createShorthand')
+      createSlotFactory(as, () => ({}))('testValue', { ...options }) // clone of the options
+
+      const optionsArg = createShorthandSpy.calls.mostRecent().args[3]
+      expect(optionsArg).toEqual({ defaultProps: { as }, ...options })
+    })
+
+    it('overrides "as" prop in defaultProps', () => {
+      const as = 'span'
+      const asOverride = 'p'
+      const createShorthandSpy = spyOn(lib, 'createShorthand')
+      createSlotFactory(as, () => ({}))('testValue', { defaultProps: { as: asOverride } }) // clone of the options
+
+      const optionsArg = createShorthandSpy.calls.mostRecent().args[3]
```

Comment:
this test is very brittle - if the next day we will check the arguments that `createShorthand` accepts, this will fail this test - while this failure won't surely guarantee that the logic of `createSlotFactory` has become to be incorrect